internal/testcontext/context.go

```diff
@@ -120,11 +123,18 @@ func (ctx *Context) Dir(subs ...string) string {
 	})
 
 	dir := filepath.Join(append([]string{ctx.directory}, subs...)...)
-	_ = os.MkdirAll(dir, 0744)
+	err := os.MkdirAll(dir, 0744)
```

Comment:
Note, Dir may be called multiple times for the particular directory. I wonder, do we need t check for is exists with mkdirall

Reply:
it won't be an error but it should be suspicious why a test is trying to create the same directory more than once.

However what happen if this is used by several test cases which they don't have any relation and they use the same values for `subs`?

Reply:
The `Dir` is meant to return a directory path and create when it doesn't yet exist. It makes some code more convenient, rather than have a variable per directory.

Reply:
I understood that correctly when I read the code, I think that I didn't understand well what you meant with 

>  I wonder, do we need t check for is exists with mkdirall

Reply:
Ah nevermind found the answer in documentation. https://golang.org/pkg/os/#MkdirAll

I was wondering whether `MkdirAll` returns an error when the directory already exists, but it does not so we can mark the comment as resolved.